sdk/search/azure-search/src/main/java/com/azure/search/models/SearchDocumentsResult.java

```diff
@@ -0,0 +1,140 @@
+    @JsonProperty(value = "@search.facets", access = JsonProperty.Access.WRITE_ONLY)
+    private Map<String, List<FacetResult>> facets;
+
+    /*
+     * Continuation JSON payload returned when Azure Cognitive Search can't
+     * return all the requested results in a single Search response. You can
+     * use this JSON along with @odata.nextLink to formulate another POST
+     * Search request to get the next part of the search response.
+     */
+    @JsonProperty(value = "@search.nextPageParameters", access = JsonProperty.Access.WRITE_ONLY)
+    private SearchRequest nextPageParameters;
+
+    /*
+     * The sequence of results returned by the query.
+     */
+    @JsonProperty(value = "value", access = JsonProperty.Access.WRITE_ONLY)
+    private List<SearchResult> results;
+
+    /*
+     * Continuation URL returned when Azure Cognitive Search can't return all
+     * the requested results in a single Search response. You can use this URL
+     * to formulate another GET or POST Search request to get the next part of
+     * the search response. Make sure to use the same verb (GET or POST) as the
+     * request that produced this response.
+     */
+    @JsonProperty(value = "@odata.nextLink", access = JsonProperty.Access.WRITE_ONLY)
+    private String nextLink;
+
+    /**
+     * Get the count property: The total count of results found by the search
+     * operation, or null if the count was not requested. If present, the count
+     * may be greater than the number of results in this response. This can
+     * happen if you use the $top or $skip parameters, or if Azure Cognitive
+     * Search can't return all the requested documents in a single Search
+     * response.
+     *
+     * @return the count value.
+     */
+    public Long getCount() {
+        return this.count;
+    }
+
+    /**
+     * Get the coverage property: A value indicating the percentage of the
+     * index that was included in the query, or null if minimumCoverage was not
+     * specified in the request.
+     *
+     * @return the coverage value.
+     */
+    public Double getCoverage() {
+        return this.coverage;
+    }
+
+    /**
+     * Get the facets property: The facet query results for the search
+     * operation, organized as a collection of buckets for each faceted field;
+     * null if the query did not include any facet expressions.
+     *
+     * @return the facets value.
+     */
+    public Map<String, List<FacetResult>> getFacets() {
+        return this.facets;
+    }
+
+    /**
+     * Get the nextPageParameters property: Continuation JSON payload returned
+     * when Azure Cognitive Search can't return all the requested results in a
+     * single Search response. You can use this JSON along with @odata.nextLink
+     * to formulate another POST Search request to get the next part of the
+     * search response.
+     *
+     * @return the nextPageParameters value.
+     */
+    public SearchRequest getNextPageParameters() {
```

Comment:
We should not expose `getNextPageParameters` or `getNextLink` publicly since they are implementation details of paging. Customers don't need to use them since they can just use `PagedIterable` and friends.

We need to fix this before preview (probably with regex transforms in the codegen config).

Reply:
since this is a generated class, all changes should be done in the autorest config and the code need to be re-generated